webui/react/src/pages/InteractiveTask.tsx

```diff
@@ -85,25 +84,25 @@ export const InteractiveTask: React.FC = () => {
         <div className={css.barContainer}>
           <TaskBar
             handleViewLogsClick={() => setPageView(PageView.TASK_LOGS)}
-            id={taskId}
-            name={taskName}
-            resourcePool={taskResourcePool}
-            type={taskType}
+            id={taskId!}
```

Comment:
im thinking to `assert` these cuz using `??` is not appropriate. Any other better options?

Reply:
**question:** correct me if I'm wrong, but there's no guarantee that `taskId` is always defined right? you can still go to routes like `/wait/notebook/` (instead of `/wait/notebook/<some-id>`. Variables coming through URL params should probably be treated as conditional definition. So `??` seems appropriate, or not showing the respective component if it requires the prop to be defined...

Reply:
It used be like [this](https://github.com/determined-ai/determined/blob/2da2ed2a1f11a87bd5999200561671149bd10cc9/webui/react/src/pages/InteractiveTask.tsx#L12). I just followed the previous style which doesn't have `?`.